sae.c

```diff
@@ -483,9 +497,13 @@ confirm_to_peer (struct candidate *peer)
     struct ieee80211_mgmt_frame *frame;
     size_t len = 0;
     BIGNUM *x, *y;
-    HMAC_CTX ctx;
+    HMAC_CTX *ctx;
     unsigned short send_conf;
     int offset;
+    ctx = HMAC_CTX_new();
+    if (ctx == NULL) {
+        return -1;
+    }
```

Comment:
This ctx will also leak if there are any errors, need to add a free in all the error paths or (probably cleaner) make a goto label at the end to cleanup the EC points x,y and the hmac context

Reply:
will fix